shared/src/models/utils/transfer-position.ts

```diff
@@ -0,0 +1,17 @@
+import { getCreate } from './get-create';
+import type { Transfer } from './transfer';
+
+export class TransferPosition {
+    public readonly type = 'transfer';
+
+    public readonly transfer: Transfer;
```

Comment:
What about flattening this? So instead of an extra property, this class would extend `Transfer` (until #612, when `TransferPosition` would completely replace `Transfer`).

Reply:
I would defer this to #612 completely